js/pro/hitbtc3.js

```diff
@@ -0,0 +1,1190 @@
+    handleOrderBookUpdate (client, message) {
+        //
+        //      {
+        //          "ch": "orderbook/full",
+        //          "update": {
+        //              "ETHBTC": {
+        //                  "t": 1626866578902,
+        //                  "s": 27617208,
+        //                  "a": [
+        //                      ["0.060508", "0"],
+        //                      ["0.060509", "2.5486"]
+        //                  ],
+        //                  "b": [
+        //                      ["0.060501", "3.9000"],
+        //                      ["0.060500", "3.0459"]
+        //                  ]
+        //              }
+        //          }
+        //      }
+        //
+        const ch = this.safeValue (message, 'ch');
+        const update = this.safeValue (message, 'update', {});
+        const keys = Object.keys (update);
+        for (let i = 0; i < keys.length; i++) {
+            const marketId = keys[i];
+            const data = update[marketId];
+            const market = this.safeMarket (marketId);
+            const symbol = market['symbol'];
+            if (symbol in this.orderbooks) {
+                const timestamp = this.safeInteger (data, 't');
+                const nonce = this.safeInteger (data, 's');
+                const orderbook = this.orderbooks[symbol];
+                const asks = this.safeValue (data, 'a', []);
+                const bids = this.safeValue (data, 'b', []);
+                this.handleDeltas (orderbook['asks'], asks);
+                this.handleDeltas (orderbook['bids'], bids);
+                orderbook['timestamp'] = timestamp;
+                orderbook['datetime'] = this.iso8601 (timestamp);
+                orderbook['nonce'] = nonce;
+                this.orderbooks[symbol] = orderbook;
+                const messageHash = ch + ':' + marketId;
+                client.resolve (orderbook, messageHash);
+            }
+        }
+    }
+
+    handleDelta (bookside, delta) {
+        const price = this.safeFloat (delta, 0);
+        const amount = this.safeFloat (delta, 1);
+        bookside.store (price, amount);
+    }
+
+    handleDeltas (bookside, deltas) {
+        for (let i = 0; i < deltas.length; i++) {
+            this.handleDelta (bookside, deltas[i]);
+        }
+    }
+
+    async watchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * get the list of most recent $trades for a particular $symbol
+         * @param {string} symbol unified $symbol of the market to fetch $trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of $trades to fetch
+         * @param {array} params extra parameters specific to the hitbtc api endpoint
+         * @return {[array]} a list of ~@link https://docs.ccxt.com/en/latest/manual.html?#public-$trades trade structures~
+         */
+        const methodParams = {
```

Comment:
We allow users to pass both the marketId and unified symbol as the symbol. Therefore it's good to make sure the symbol is the unified version if used later in the function.
```suggestion
        await this.loadMarkets ();
        const market = this.market (symbol);
        symbol = market['symbol'];
        const methodParams = {
```